js/main.js

```diff
@@ -13,11 +13,45 @@ $avatarUrl.addEventListener('input', function (event) {
 
 $formSelect.addEventListener('submit', function (event) {
   event.preventDefault();
-  data.profile.username = $userNameInfo.value;
-  data.profile.avatarUrl = $avatarUrl.value;
-  data.profile.fullName = $userFullNameInfo.value;
-  data.profile.location = $userLocationInfo.value;
-  data.profile.bio = $userBioInfo.value;
+  if (!$userNameInfo.value.trim()) { // prevents user from having username with spaces
+    document.querySelector('.container').classList.add('modal-check');
```

Comment:
This line and the line below it also would be a great candidate for a function named something like `displaySubmitErrorModal`:grin: